scripts/merge-script/cli.ts

```diff
@@ -0,0 +1,183 @@
+import chalk from 'chalk';
+import {prompt} from 'inquirer';
+import * as minimist from 'minimist';
+import {isAbsolute, resolve} from 'path';
+import {Config, readAndValidateConfig} from './config';
+
+import {
+  MergeResult,
+  MergeStatus,
+  PullRequest,
+  PullRequestFailure,
+  PullRequestMergeTask,
+  PullRequestStatus
+} from './index';
+
+main();
+
+/** List of pull request statuses which are not critical and can be forcibly ignored. */
+const FORCE_OVERRIDABLE_FAILURES = [
+  PullRequestStatus.PENDING_CI_JOBS,
+  PullRequestStatus.FAILING_CI_JOBS,
+];
+
+/**
+ * Entry-point for the merge script CLI. The script can be used to merge multiple pull requests
+ * into branches based on the `PR target` labels that have been set in a configuration. The script
+ * aims to reduce the manual work that needs to be performed to cherry-pick a PR into multiple
+ * branches based on a target label.
+ */
+async function main() {
+  const {config, pullRequests, force, dryRun, githubToken: cliGithubToken} = parseCommandLine();
+  const githubToken = cliGithubToken || process.env.GITHUB_TOKEN || process.env.TOKEN;
+
+  if (!githubToken) {
+    console.error(
+        chalk.red('No Github token is set. Please set the `GITHUB_TOKEN` environment variable.'));
+    console.error(chalk.red('Alternatively, pass the `--github-token` command line flag.'));
+    process.exit(1);
+  }
+
+  const api = new PullRequestMergeTask(config, githubToken);
+
+  // Merge the pull request and handle the merge result.
+  await handleMergeResult(await api.merge(pullRequests, dryRun, false), force);
+
+  /**
+   * Prompts whether failed pull requests should be forcibly merged. If so, merges
+   * the specified failed pull requests forcibly (ignoring non-critical failures).
+   */
+  async function promptAndPerformForceMerge(failedPrs: number[]) {
+    if (await promptConfirm('Do you want to forcibly proceed with merging?')) {
+      // Perform a merge in force mode and handle the merge result. Note that we disable the
+      // force merge prompt since a failed force merge will never succeed with a second force merge.
+      await handleMergeResult(await api.merge(failedPrs, dryRun, true), true);
+    }
+  }
+
+  /**
+   * Handles the merge result by printing console messages, exiting the process
+   * based on the result, or by restarting the merge if force mode has been enabled.
+   */
+  async function handleMergeResult(
+      {failures, status, merged}: MergeResult, disableForceMergePrompt = false) {
+    const canForciblyMerge = failures ?
+        failures.every(failure => FORCE_OVERRIDABLE_FAILURES.includes(failure.status)) :
+        false;
+
+    if (status === MergeStatus.SUCCESS) {
```

Comment:
I personally find this more readable, with `if` blocks that have returns since we are also either exiting the function or the entire process on each match.

Reply:
I converted it to a `switch` statement. Let me know if that is better.